bundles/org.openhab.binding.bluelink/src/main/resources/OH-INF/thing/vehicle.xml

```diff
@@ -0,0 +1,308 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="bluelink"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="vehicle">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="account"/>
+		</supported-bridge-type-refs>
+
+		<label>Bluelink Vehicle</label>
+		<description>A Hyundai vehicle connected via Bluelink.</description>
+
+		<channel-groups>
+			<channel-group id="status" typeId="status"/>
+			<channel-group id="charging" typeId="charging"/>
+			<channel-group id="fuel" typeId="fuel"/>
+			<channel-group id="doors" typeId="doors"/>
+			<channel-group id="windows" typeId="windows"/>
+			<channel-group id="climate" typeId="climate"/>
+			<channel-group id="range" typeId="range"/>
+			<channel-group id="warnings" typeId="warnings"/>
+		</channel-groups>
+
+		<properties>
+			<property name="vin">Vehicle Identification Number</property>
+			<property name="model">Vehicle Model</property>
+			<property name="engineType">Engine Type</property>
+		</properties>
+
+		<representation-property>vin</representation-property>
+
+		<config-description>
+			<parameter name="vin" type="text" required="true">
+				<label>VIN</label>
+				<description>Vehicle Identification Number (VIN).</description>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" min="0" unit="min" required="false">
+				<label>Refresh Interval</label>
+				<description>Interval in minutes to fetch cached data from the Hyundai/Kia servers.</description>
+				<default>30</default>
+			</parameter>
+			<parameter name="forceRefreshInterval" type="integer" min="0" unit="min" required="false">
+				<label>Refresh Interval</label>
+				<description>Interval in minutes to refresh data from the vehicle.</description>
+				<default>240</default>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- Channel Group Types -->
+
+	<channel-group-type id="status">
+		<label>Vehicle Status</label>
+		<channels>
+			<channel id="locked" typeId="locked"/>
+			<channel id="engine-running" typeId="engine-running"/>
+			<channel id="odometer" typeId="odometer"/>
+			<channel id="battery-level" typeId="battery-level"/>
+			<channel id="location" typeId="system.location"/>
+			<channel id="last-update" typeId="last-update"/>
+		</channels>
+	</channel-group-type>
+
+	<channel-group-type id="charging">
+		<label>Charging</label>
+	</channel-group-type>
```

Comment:
Empty channel group?

Reply:
Populated with dynamic channels only for electric vehicles. Should I remove this?

Reply:
Or maybe they are already added dynamically. Unable to look at the code from my phone now

Reply:
I wasn't able to figure out how to add channel groups progammatically. However the empty groups do not appear in the UI as far as I can tell, so it seems harmless.